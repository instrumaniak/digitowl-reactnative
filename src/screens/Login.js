/**
 *   Login/Register screen
 */

import React from 'react'
import PropTypes from 'prop-types'

import {
  View,
  StyleSheet
} from 'react-native'

import {
  Container,
  Header,
  Title,
  Left,
  Right,
  Body,
  Content,
  Button,
  Text,
  Icon,
  Form,
  Item,
  Input
} from 'native-base'

import { colors } from '../config'

const Login = ({ navigation }) => {
  const type = navigation.getParam('type', '')
  const buttonText = type == 'register' ? 'Signup' : 'Login'
  return (
    <Container>
      <Content padder>
        <View style={{paddingBottom: 0}}>
          <Form style={{ marginBottom: 20}}>
            <Item>
              <Icon name='md-mail' style={styles.icons}/>
              <Input placeholder='Email' />
            </Item>
            <Item>
              <Icon name='md-lock' style={styles.icons}/>
              <Input placeholder='Password' secureTextEntry/>
            </Item>
          </Form>
          <Button block style={{backgroundColor: colors.greenDark}}>
            <Text>Continue</Text>
          </Button>
        </View>

        <View>
          <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
            <View style={styles.dash} />
            <Text style={styles.textOr}>or</Text>
            <View style={styles.dash} />
          </View>
          <Button iconLeft block>
            <Icon name='logo-facebook' style={styles.icons}/>
            <Text uppercase={false}>{buttonText} with Facebook</Text>
          </Button>
          <Button block iconLeft light style={{ marginTop: 15 }}>
            <Icon name='logo-google' style={styles.icons}/>
            <Text uppercase={false}>  {buttonText} with Google   </Text>{/* Space for alignment */}
          </Button>
        </View>

        <View style={{marginTop: 180}}>
          <Text style={styles.notice}>By using DigitOwl you agree with DigitOwl{"'"}s {' '}
            <Text style={styles.link} onPress={() => alert('Terms')}>terms of use</Text> & {' '}
            <Text style={styles.link} onPress={() => alert('Privacy')}>privacy policy</Text>.</Text>
        </View>
      </Content>
    </Container>
  )
}

// Header
Login.navigationOptions = ({ navigation }) => ({
  header: (
    <Header androidStatusBarColor={colors.greenDark} style={{ backgroundColor: colors.green}}>
      <Left>
        <Button transparent onPress={() => navigation.goBack()}>
          <Icon name='arrow-back' />
        </Button>
      </Left>
      <Body>
        <Title>{ navigation.getParam('type') == 'register' ? 'Create account' : 'Login'}</Title>
      </Body>
      <Right />
    </Header>
  )
})

Login.propTypes = {
  navigation: PropTypes.any
}

const styles = StyleSheet.create({
  icons: {
    color: colors.grey
  },
  textOr: {
    padding: 20,
    textAlign: 'center',
    fontStyle: 'italic'
  },
  notice: {
    padding: 10,
    fontSize: 14,
    fontStyle: 'italic',
    color: colors.grey
  },
  link: {
    fontSize: 14,
    color: colors.grey,
    textDecorationLine: 'underline'
  },
  dash: {
    height: 1,
    width: 100,
    backgroundColor: colors.grey
  }
})

export default Login
