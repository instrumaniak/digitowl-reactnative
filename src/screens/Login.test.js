/**
 *   Tests for Login component
 */

import React from 'react'
import Login from './Login'
import renderer from 'react-test-renderer'

describe('Login Screen', () => {
  it('Renders correctly', () => {
    // mock react navigation
    const navigation = {
      navigate: jest.fn(),
      getParam: jest.fn()
    }

    const tree = renderer.create(
      <Login navigation={navigation}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
