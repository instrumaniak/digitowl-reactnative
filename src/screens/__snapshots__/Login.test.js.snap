// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": 10,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    padder={true}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "marginBottom": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 15,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                      "paddingRight": 8,
                    },
                    Object {
                      "color": "#5B6267",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              editable={true}
              placeholder="Email"
              placeholderTextColor="#575757"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "top": 1.5,
                }
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#D9D5DC",
                "borderLeftWidth": 0,
                "borderRadius": undefined,
                "borderRightWidth": 0,
                "borderTopWidth": 0,
                "borderWidth": undefined,
                "flexDirection": "row",
                "marginLeft": 15,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 24,
                      "paddingRight": 8,
                    },
                    Object {
                      "color": "#5B6267",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              editable={true}
              placeholder="Password"
              placeholderTextColor="#575757"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "top": 1.5,
                }
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#119DA4",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Continue
          </Text>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#c3c6c7",
                "height": 1,
                "width": 100,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontStyle": "italic",
                  "padding": 20,
                  "textAlign": "center",
                },
              ]
            }
            uppercase={false}
          >
            or
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#c3c6c7",
                "height": 1,
                "width": 100,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#f4f4f4",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                  Object {
                    "color": "#5B6267",
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "#4B5358",
                },
              ]
            }
            uppercase={false}
          >
            Login
             with Facebook
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#f4f4f4",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 15,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                  Object {
                    "color": "#5B6267",
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "#4B5358",
                },
              ]
            }
            uppercase={false}
          >
              
            Login
             with Google   
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 180,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#c3c6c7",
                "fontSize": 14,
                "fontStyle": "italic",
                "padding": 10,
              },
            ]
          }
          uppercase={false}
        >
          By using DigitOwl you agree with DigitOwl
          '
          s 
           
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "#c3c6c7",
                  "fontSize": 14,
                  "textDecorationLine": "underline",
                },
              ]
            }
            uppercase={false}
          >
            terms of use
          </Text>
           & 
           
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "#c3c6c7",
                  "fontSize": 14,
                  "textDecorationLine": "underline",
                },
              ]
            }
            uppercase={false}
          >
            privacy policy
          </Text>
          .
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
